nexus/src/app/switch_port.rs

```diff
@@ -289,6 +296,48 @@ impl super::Nexus {
 
         Ok(())
     }
+
+    pub(crate) async fn switch_port_status(
+        &self,
+        opctx: &OpContext,
+        switch: Name,
+        port: Name,
+    ) -> Result<SwitchLinkState, Error> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        let loc: SwitchLocation = switch.as_str().parse().map_err(|e| {
+            Error::invalid_request(&format!(
+                "invalid switch name {switch}: {e:?}"
+            ))
+        })?;
+
+        let dpd_clients = self.dpd_clients().await.map_err(|e| {
+            Error::internal_error(&format!("dpd clients get: {e}"))
+        })?;
+
+        let dpd = dpd_clients.get(&loc).ok_or(Error::internal_error(
+            &format!("no client for switch {switch}"),
+        ))?;
+
+        let port_id = PortId::Qsfp(port.as_str().parse().map_err(|e| {
+            Error::invalid_request(&format!("invalid port name: {port} {e}"))
+        })?);
```

Comment:
should be moved up to above `dpd_clients` -- input validation should happen first.